Taxing of lines without tax class (SHOOP-2077)

suutari-ai opened the pull request:
Implement taxing of lines without a tax class, especially discount
lines.

Has some other related changes too.  See the commit messages for
details.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1876/
Test FAILed.

tulimaki commented on shoop_tests/functional/test_discount_taxing.py:
```diff
@@ -0,0 +1,544 @@
+from collections import defaultdict
```
Add ``from __future__ import unicode_literals`` for python 2 also you might want to add the license header.

suutari-ai replied:
Will do. Thanks.

tulimaki commented on shoop/core/taxing/utils.py:
```diff
@@ -6,13 +6,53 @@
+
+    :type lines: list[shoop.core.order_creator.SourceLine]
+    :param lines: List of taxed lines to generate proportions from
+
```
Extra empty line

suutari-ai replied:
Actually it was intentional, to separate input params from return param, but it seems that other docstrings don't have a line break there, so I'll remove this too. :)

tulimaki commented on shoop_tests/functional/test_discount_taxing.py:
```diff
@@ -0,0 +1,544 @@
+        assert get_pretty_tax_summary(source) == [
+            'Tax-A   0.25  180.000000000   45.000000000  225.000000000']
+        assert get_pretty_line_taxes_of_discount_lines(source) == [[
+            'Tax-A   0.25  -20.000000000   -5.000000000  -25.000000000']]
```
Based on this. Should the line total be -20. If merchant use taxless prices in shop, is the only way to set discount amount 20 to actually calculate possible taxes. Sould the discount amount in campaings follow shop prices whether is taxless or taxfull.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1891/
Test FAILed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1892/
Test FAILed.

suutari-ai commented:
Fixed the unicode issue in the test module and added the license header. Removed the extra blank from docstring of `get_tax_class_proportions`. Also rebased to current master, since had conflicts in changelog.

suutari-ai commented:
retest this please

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1893/
Test PASSed.
